compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -523,7 +529,9 @@ object Denotations {
               try infoMeet(info1, info2)
               catch {
                 case ex: MergeError =>
-                  if (pre.widen.classSymbol.is(Scala2x) || ctx.scala2Mode)
+                  if (preferSym(sym2, sym1)) info2
```

Comment:
Would it be difficult to construct a test case to exercise this? Also, are we confident this is sound?

Reply:
I tried for a while but did not manage. The code that breaks is in class `Id` in `MapView` of the collection strawman.

Reply:
Regarding soundness: What happens is that we get two denotations that should be overloaded and we pick one and forget about the other. I see that as a possible problem for completeness, not sure how it would affect soundness.

Reply:
Maybe add a comment to note that this may be worth revisiting at some point?